src/js/form.js

```diff
@@ -313,26 +328,32 @@ Form.prototype.init = function () {
 
         // after radio button data-name setting (now done in XLST)
         // Set temporary event handler to ensure calculations in newly added repeats are run for the first time
-        const tempHandlerAddRepeat = (event) => this.calc.update(event.detail);
-        const tempHandlerRemoveRepeat = () => (this.all = {});
+        let repeatsAdded = 0;
+
+        const tempHandlerAddRepeat = () => {
+            repeatsAdded++;
+        };
         this.view.html.addEventListener(
             events.AddRepeat().type,
             tempHandlerAddRepeat
         );
-        this.view.html.addEventListener(
-            events.RemoveRepeat().type,
-            tempHandlerRemoveRepeat
-        );
         this.repeatsInitialized = true;
         this.repeats.init();
+
+        // Recalculate after repeats are initialized. Previously this was performed in
+        // `tempHandlerAddRepeat`, but recalculating them all at once is significantly faster.
```

Comment:
I would also separate this mostly because I think the asynchronous evaluation might turn out to be a very significant change and should be in its own commit without distractions.